luigi/retcodes.py

```diff
@@ -82,6 +82,11 @@ def run_with_retcodes(argv):
 
     task_sets = luigi.execution_summary._summary_dict(worker)
     root_task = luigi.execution_summary._root_task(worker)
+    if root_task in task_sets['completed']:
+        completed_tasks = task_sets['completed']
+        for key in task_sets.keys():
+            if key != 'completed':
+                task_sets[key] = task_sets[key] - completed_tasks
```

Comment:
Do you really need all this?

I'm pretty sure you can just remove all this, and below just check `(retcodes.task_failed, has('failed'))` instead of `has('ever_failed'). No? I thought we introduced `ever_failed` and defined `failed` to mean something that in the very end was count as failed.